app/controllers/api/v1/npq/users_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module NPQ
+      class UsersController < Api::ApiController
+        include ApiTokenAuthenticatable
+
+        def create
+          creation_response = ::NPQ::Users::FindOrCreateBy.new(params: find_or_create_params).call
```

Comment:
It alternatively returns an OpenStruct containing errors in case of failure, rather than one containing a User.

But you're right, an alternative where it just always returns the user and that may have errors is possible. I preferred the hard distinction between the user and errors so that `creation_response.users.present?` always determines success or failure